### PR TITLE
Align ddp and ddp-spawn strategies in setting up the environment

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Aligns the regular ddp strategy and the ddp spawn strategy in setting up the environment (through the method `setup_environement` which is called in the Trainer._run).

**Before:**

DDP: Setup environment in `setup_environment` method/hook.
DDPSpawn: At spawn time, in worker_setup function (slightly earlier than `setup_environment` method/hook

**Now:**

Both DDP and DDPSpawn set the same variables in `setup_environment` method.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @awaelchli @rohitgr7 @kaushikb11 @akihironitta

### CODE REVIEW

**Comment by ananthsub** [comment]
I believe this will be a prerequisite for further consolidation efforts like https://github.com/PyTorchLightning/pytorch-lightning/pull/11021 and https://github.com/PyTorchLightning/pytorch-lightning/pull/11020 by moving up the `setup_environment` of DDP and TPU spawn

**Comment by tchaton** [comment]
Hey @awaelchli. What is the status on this PR ?

**Comment by awaelchli** [comment]
The status is that I'm currently working on this and once #11163 is merged this will pretty much be ready.

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
@awaelchli is this important for Lite? Do you want to apply these changes to the Lite versions already in #14670 and #14675?

**Comment by awaelchli** [comment]
@carmocca I would say quite important for #14707. Yes, I can integrate this into the two PRs. The PR was ready long before starting with Lite, I should have pushed harder to get this merged 😅

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed fall-back to `LightningEnvironment` when number of SLURM tasks does not correspond to number of processes in Trainer ([#14300](https://github.com/Lightning-AI/lightning/pull/14300))
 
 
+- Aligned DDP and DDPSpawn strategies in setting up the environment ([11073](https://github.com/Lightning-AI/lightning/pull/11073))
```
Comment (krshrimali) [inline src/pytorch_lightning/CHANGELOG.md:77]:
```suggestion
- Aligned DDP and DDPSpawn strategies in setting up the environment ([#11073](https://github.com/Lightning-AI/lightning/pull/11073))
```

**Comment by Borda** [comment]
@awaelchli mind checking the conflicts... :otter:

File: src/pytorch_lightning/strategies/tpu_spawn.py [hunk 2]
```diff
@@ -212,9 +212,9 @@ def reduce(
 
         return output
 
-    def _worker_setup(self, process_idx: int) -> None:
+    def setup_distributed(self) -> None:
```
Comment (carmocca) [inline src/pytorch_lightning/strategies/tpu_spawn.py:215]: Should this be `setup_environment`?

This implementation matches exactly `XLAStrategy.setup_environment`: https://github.com/Lightning-AI/lightning/blob/7ff297bdffbc53beb6ffc999c013341d3a0d09c8/src/lightning_lite/strategies/xla.py#L97-L100

Right now, this difference between strategies is causing https://github.com/Lightning-AI/lightning/pull/14926#discussion_r983009669

Alternatively, if we want `setup_distributed` here then `XLAStrateg`y should change `setup_environment -> setup_distributed` too.

Comment (awaelchli) [reply src/pytorch_lightning/strategies/tpu_spawn.py:215]: `setup_environment` calls `setup_distributed` and does nothing else except calling `super().setup_environment`. 

This may look weird to you, why not collapse the two similar methods into one? 

The answer is very simple: It wasn't always like this. For historical reasons, these two methods had to handle different things, and they existed also to be overridden differently in the subclasses, e.g. TPUSpawn. Overtime, we have simplified things, but it is a slow process and the strategies are just in a constant state of incompleteness. 

In summary, this PR makes 

`setup_environment` in DDP and DDPSpawn identical, because they should be!
`setup_distributed` in DDP and DDPSpawn identical, because they should be!


 Maybe this helps in https://github.com/Lightning-AI/lightning/pull/14926#discussion_r983009669, but I'm sure the accelerator issue is unrelated. I will take a look there.

Comment (carmocca) [reply src/pytorch_lightning/strategies/tpu_spawn.py:215]: Resolving, we did

> Alternatively, if we want setup_distributed here then XLAStrategy should change setup_environment -> setup_distributed too.

**Review by justusschock — APPROVED** [review]

**Comment by carmocca** [comment]
:tada:

**Comment by awaelchli** [comment]
🥳
